tests/ext-popup.test.js

```diff
@@ -23,187 +26,205 @@ function observeChange(element) {
   document.body.innerHTML = popupBody;
 
-  const getMonitorStatusFn = jest.spyOn(ExtListen, 'getMonitorStatus');
-  const stopMonitorFn = jest.spyOn(ExtListen, 'stopMonitor');
+  // assume, extensions are being monitored
+  getMonitorStatusSpyFn.mockResolvedValueOnce(true);
+  // extensions are not being monitored after the stop monitoring button is clicked
+  getMonitorStatusSpyFn.mockResolvedValueOnce(false);
+  stopMonitorSpyFn.mockResolvedValue(true);
 
   const popup = new Popup();
+  await popup.init();
 
   const startBtn = popup.startMonitorAllBtn;
   const stopBtn = popup.stopMonitorAllBtn;
   const monitorStatus = popup.monitorStatusText;
   const errorText = popup.errorMsgText;
 
-  getMonitorStatusFn.mockResolvedValueOnce(true);
-  getMonitorStatusFn.mockResolvedValueOnce(false);
-  stopMonitorFn.mockResolvedValue(true);
-
-  await popup.init();
-
-  expect(startBtn.hasAttribute('disabled')).toBeTruthy();
-  expect(stopBtn.hasAttribute('disabled')).toBeFalsy();
-
-  expect(monitorStatus.textContent).toBe('Extensions are being monitored');
-  expect(monitorStatus.classList.contains('failure')).toBeFalsy();
-  expect(monitorStatus.classList.contains('success')).toBeTruthy();
-
-  expect(errorText.textContent).toBe('');
-
-  const promiseElementChanged = observeChange(monitorStatus);
+  const observeMonitorStatusChanged = observeChange(monitorStatus);
   stopBtn.click();
-  await promiseElementChanged;
+  await observeMonitorStatusChanged;
 
   expect(startBtn.hasAttribute('disabled')).toBeFalsy();
   expect(stopBtn.hasAttribute('disabled')).toBeTruthy();
 
-  expect(monitorStatus.textContent).toBe('No extensions are being monitored');
+  expect(monitorStatus.textContent).toBe(MONITOR_DISABLED_MSG);
   expect(monitorStatus.classList.contains('failure')).toBeTruthy();
   expect(monitorStatus.classList.contains('success')).toBeFalsy();
   expect(errorText.textContent).toBe('');
+});
 
-  getMonitorStatusFn.mockClear();
+test('clicking on "view activity logs page" button should open a new tab with activity logs page', async () => {
+  document.body.innerHTML = popupBody;
+
+  const ACTIVITY_LOG_PAGE_URL =
+    'moz-extension://extension-page/activitylog/activitylog.html';
+
+  const getURL = jest.fn().mockReturnValue(ACTIVITY_LOG_PAGE_URL);
+  const create = jest.fn();
+
+  window.browser = {
+    runtime: { getURL },
+    tabs: { create },
+  };
+
+  // prevent window.close to trigger a failure due to jest teardown not expecting the jsdom window to be gone
+  window.close = jest.fn(() => {});
+
+  const popup = new Popup();
+  await popup.init();
+
+  popup.openActivityLogBtn.click();
+
+  expect(getURL).toHaveBeenCalled();
+  expect(create).toHaveBeenCalledWith({ url: ACTIVITY_LOG_PAGE_URL });
+
+  setImmediate(() => {
+    // expect the browserAction popup to autoclose itself.
+    expect(window.close).toHaveBeenCalled();
+  });
 });
 
-test('errors on start/stopping monitoring should be shown in the popup UI', async () => {
```

Comment:
this test has been split into two tests, one is  `error at start monitoring extensions should be displayed` and another one is `error at stop monitoring extensions should be displayed`. This made the test code more aligned for its single purpose.